### DISCUSSION TITLE
pagsurface创建崩溃

### DISCUSSION THREAD

**Discussion by qingpingwang** [Tencent/libpag#2416] [General] [answered]
**使用场景**
本人使用场景需要获取OpenGL的上下文，做一些额外的渲染工作。所以需要借助pagsurface获取到OpenGL上下文。
**复现步骤**
1.创建离屏pagsurface；
2.调用lockContext()与unlockContext();
3.循环重复1、2；
**运行环境**
macos 13.4
**代码截图**
![20240812-144427](https://github.com/user-attachments/assets/9325554f-2a91-4045-a0df-b224c10eb951)

**Comment by qingpingwang** [comment]
创建多次后会崩溃。

**Comment by qingpingwang** [comment]
"Context leak detected, msgtracer returned -1"，平均30次创建与销毁会出现一次。

**Comment by qingpingwang** [comment]
![image](https://github.com/user-attachments/assets/d8e9c99e-d1b8-4ea3-8d1a-cc88eeb6d893)

**Comment by qingpingwang** [comment]
`
auto surface = pag::PAGSurface::MakeOffscreen(1080, 1920);
if (surface == nullptr) {std::cout << "---pagSurface is nullptr!!!" << std::endl;}
surface->clearAll();
`
以上代码运行效果也一样

**Comment by kevingpqi123** [comment, marked as answer]
提供信息不全，建议将背景、使用版本都说清楚，这里不会有问题，pag 的每一次提交都会走自动化测试用例，使用的都是PAGSurface::MakeOffscreen的接口
另外，macOS 平台获取 PAGSuface 的 OpenGL 上下文做什么用呢，本身就可以获取到 CVPixelBuffer 的渲染数据， 直接操作 CVPixelBuffer 就可以
